server/helper.ts

```diff
@@ -44,14 +44,27 @@ function getVersionNumber() {
 	return pkg.version;
 }
 
+let _fetchedGitCommit = false;
 let _gitCommit: string | null = null;
 
 function getGitCommit() {
-	if (_gitCommit) {
+	if (_fetchedGitCommit) {
 		return _gitCommit;
 	}
 
-	if (!fs.existsSync(path.resolve(__dirname, "..", ".git"))) {
+	_fetchedGitCommit = true;
+
+	let rootPath = __dirname;
+
+	// Typescript build will be in dist/ folder,
+	// but it will not be when running `dev`.
+	if (rootPath.endsWith(path.join("dist", "server"))) {
+		rootPath = path.basename(rootPath);
+	}
+
+	// Check that .git folder exists in the expected place,
+	// to avoid git command mistakenly recursing outside of our root folder
+	if (!fs.existsSync(path.resolve(rootPath, "..", ".git"))) {
```

Comment:
Note that this isn't necessarily a folder... so that comment is wrong.
If it's a worktree, .git is a file.

Reply:
TIL, but that doesn't strictly matter.

Reply:
I know, but I already had to patch out the git folder detection once. And I'd like to avoid having to do so again so I'd appreciate it if we can get rid of the notion that .git is a folder

Reply:
thanks